### ISSUE TITLE
Uninitialized constant Digest::SHA1

### ISSUE THREAD

**Issue by gogainda** [oracle/truffleruby#2577]
Intalling this Gemfile fails on Truffleruby:
```

source "https://rubygems.org"

gem "bootsnap"
gem "ruby-prof", platform: :ruby
gem "pry"
gem "pry-stack_explorer", platform: :ruby
gem "rails", github: "rails/rails", require: "rails/all", ref: "main"
gem 'sqlite3', "~> 1.4", platform: :ruby
gem "activerecord-jdbcsqlite3-adapter", platform: :jruby
gem "sequel"
gem "evt"
if RUBY_ENGINE == "ruby" # This doesn't work on truffle-ruby becuase there's no `IO::READABLE`
  gem "libev_scheduler"
end
```
issue initially reported here https://github.com/rmosolgo/graphql-ruby/pull/3852

**Comment by eregon** [comment]
Fixed in c1655b30e3461997f1ebab976427e4789a319b79

**Comment by eregon** [comment]
To improve search, here are some examples errors this issue could cause, they are all solved now in dev builds and will be in the 22.1 release:
```
NameError: uninitialized constant Digest::MD5
NameError: uninitialized constant Digest::SHA1
NameError: uninitialized constant Digest::SHA2
NameError: uninitialized constant Digest::SHA256
NameError: uninitialized constant Digest::SHA384
NameError: uninitialized constant Digest::SHA512
```

The problem is typically caused by `digest` becoming a default gem and CRuby adding it as a dependency of net-imap/net-pop/net-smtp: https://github.com/ruby/net-protocol/pull/6 (even though it's not needed)
